Source/Core/Common/TraversalClient.h

```diff
@@ -0,0 +1,87 @@
+// This file is public domain, in case it's useful to anyone. -comex
+
+#pragma once
+
+#include "Common.h"
+#include "Thread.h"
+#include "TraversalProto.h"
+#include <functional>
+#include <memory>
+#include <list>
+#include "enet\enet.h"
+
+class TraversalClientClient
+{
+public:
+	virtual void OnTraversalStateChanged()=0;
```

Comment:
Should also add a virtual destructor here for the future as a just in case, since it's an interface.